### ISSUE TITLE
Only automatically register a data-store once-per-import.

### ISSUE THREAD

**Issue by tofumatt** [google/site-kit-wp#2223]
## Bug Description

Our data stores currently have the side-effect of registering themselves to the default data store when loaded. This means that separate entrypoints loading (other imports, likely) from this file can cause the store to be re-registered.

This was an issue somewhat solved by #2145, but https://github.com/google/site-kit-wp/issues/2145#issuecomment-713475587 shows there's still more potential for errors.

WordPress/Gutenberg gets around this by storing actions, selectors, etc. all in their own files, but that organisational approach makes for less maintainable/readable stores (we've discussed this in the past).

Going forward, we should only register the store if an existing one with the same name isn't found—we can do this by checking `Data.stores[ STORE_NAME ]` when auto-registering, or by saving each activated store in a hidden global in `Data._autoActivatedStores`. The latter is an extra variable, but would give a bit more insight into _why_ a store was registered, which could be useful in debugging.

---------------

_Do not alter or remove anything below. The following sections will be managed by moderators only._

## Acceptance criteria

* A data store should only ever be automatically registered to the global `googlesitekit.data`/`Data` instance once.
* Importing `actions`, `selectors`, etc. from a store that has already been imported should not cause another store to be registered, as mentioned in https://github.com/google/site-kit-wp/issues/2145#issuecomment-713475587.

## Implementation Brief

* 

### Test Coverage

* 

### Visual Regression Changes

* 

## QA Brief

* 

## Changelog entry

*

**Comment by aaemnnosttv** [comment]
@tofumatt @felixarntz this would be a non-issue if we only registered the module's data store in its entry point rather than the data store's index module. This is the direction I took with my [POC branch](https://github.com/google/site-kit-wp/pull/2593/files?diff=split&w=1) on #2587. While I'm open to other solutions, I do think that all of our code that is specific to global instances belongs in the entry point for this reason since they'll never be double initialized and that's the only place it makes sense to reference globals. The rest of the code can be more abstract and based on type/interface. As you'll see in my branch, this can still be very minimal, clean, and consistent across modules.
